asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py

```diff
@@ -100,6 +134,49 @@ def write_html(html, path):
         with open(path, "w") as f:
             f.write(html)
 
+    def swap_b_factor(self, protein, fitness_data):
+        """
+        Given a dict of fitness values, swap out the b-factors in the protein.
+        """
+        # first need to swap the protein's b-factor with fitness scores. Could do this with str.split() but gets a bit complicated.
+        parser = PDBParser()
+        protein_biopython = parser.get_structure("protein", str(protein))
+        self.logger.warning(f"Swapping b-factor with fitness score.")
+        for res in protein_biopython.get_residues():
+            res_number = res.get_full_id()[3][1]  # what a world we live in..
+            try:
+                for at in res.get_atoms():
+                    at.set_bfactor(fitness_data[res_number])
+            except KeyError:
```

Comment:
This is leaving original B factor, in we will need to fix this going forward.

Reply:
yes, currently in the case of Mpro there are 3 missing residues. We'll need to:
- store which residues are missing
- pass that info on to the HTML viz section
- in HTML blocks, make `NGL` color these residues a contrasting color (e.g. blue)

Can do in a next PR because there are so few missing residues it won't affect the viz